scripts/preprocess_data.py

```diff
@@ -0,0 +1,192 @@
+import pandas as pd
+import os, sys
+from collections import defaultdict
+
+
+data_infile = sys.argv[1]
+data_outfile = sys.argv[2]
+
+lines = []
```

Comment:
Suggest wrapping all lines below this in a `main(data_infile, data_outfile)` function and then creating a new block at the bottom of the file:
```python
if __name__ == '__main__':
    data_infile = sys.argv[1]
    data_outfile = sys.argv[2]
    main(data_infile, data_outfile)
```
This allows you to separate parsing command line arguments, and the running of the script.

Suggest also adding an error message if the incorrect arguments are passed, and returning a `sys.exit(1)` (non-zero) value if the code fails.

Reply:
Wrapped most of the script in a `main()` function which is called at the bottom of the file. 
Haven't yet added an error message though.

Reply:
Something like this:
```python
if __name__ == '__main__':

    if len(sys.argv) != 3:
        msg = "Usage: python {} <infile> <outfile>"
        print(msg.format(sys.argv[0]))
        sys.exit(1)
    else:
        data_infile = sys.argv[1]
        data_outfile = sys.argv[2]
        main(data_infile, data_outfile)
```